.github/workflows/docs-preview-deploy.yml

```
@@ -47,10 +45,10 @@ jobs:
           head_sha: ${{ github.event.workflow_run.head_sha }}
           pull_requests: ${{ tojson(github.event.workflow_run.pull_requests) }}
         run: |
-          PR_NUMBER=$(jq -r '[.[] | select(.head.sha == "${{ env.head_sha }}")][0].number' <<< '${{ env.pull_requests }}')
+          PR_NUMBER=$(jq -r '[.[] | select(.head.sha == "${head_sha}")][0].number' <<< "${pull_requests}")
```

This will still be invalid:

```bash
jq -r '[.[] | select(.head.sha == "${head_sha}")][0].number'
```

`jq` is splatting (`.[]`) all array items from `pull_requests` variable, and selecting only the one that has `.head.sha` value that matches the expected checksum, but due to single quote wrapping this will not work. Flipping the quotes alone will not work either, you'd need to escape the inner `"` AFAIK.

The `env` input appears fairly trustworthy though? I don't think the branch name can be used as an attack vector with quotes?

```suggestion
          PR_NUMBER=$(jq -r "[.[] | select(.head.sha == \"${head_sha}\")][0].number" <<< "${pull_requests}")
```

vs

```suggestion
          PR_NUMBER=$(jq -r '[.[] | select(.head.sha == "${{ env.head_sha }}")][0].number' <<< '${{ env.pull_requests }}')
```

The latter seems simpler to grok, so long as the input is not exploitable that seems less prone to human error with maintenance?

---

> I don't think the branch name can be used as an attack vector with quotes?

The branch name can, but the SHA is safe to use, so in this case its ok to use the env workflow expression